docs/DeveloperGuide.adoc

```diff
@@ -153,7 +153,175 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+For example, instead of using optionA/ to denote the first option, we use a/ instead. Although this may be unfamiliar
+to the user initially, it should be easy to pick up and will make it less tedious during input.
+
+* ** Alternatives Considered: **
+** Using more descriptive terms as arguments such that each argument will be specified clearly. However, this decreases
+the user experience as the command will be too long.
+
+[[Feature-Slideshow]]
+=== Slideshow feature
+
+The slideshow feature is dependent on the questions added by the `question slideshow [question no(s).]` command and
+interacts with the `ModelManager` to retrieve the list of questions to be displayed in the slideshow.
+The logic control for displaying the ui resides in the `SlideshowWindow` class and handles the controls and instantiation
+of the various `QuestionPanel` that contains each question. +
+
+Below is the sequence diagram of the interactions that happen from when the slideshow command is entered, to the corresponding
+questions displayed in the slideshow.
+
+image::SlideshowFeatureSequenceDiagram.png[]
+
+The following is an example and detailed explanation as to how the questions are fetched and displayed on the slideshow. +
+
+*Step 1.* The user requests to start a slideshow with a selection of questions using the `question slideshow [question no(s).]` command.
+This will add questions based on the `Index` specified and will be added to the `slideshowQuestions` list under the `SavedQuestions` class.
+The usage of `Index` here is such that it follows the same convention of when the user edits or deletes a question.
+[NOTE]
+The order of the questions displayed on the slideshow will be *based on the input order*.
+
+*Step 2.* The command is executed and the `MainWindow` calls `CommandResult#isShowSlideshow()` to verify if the command specified
+is to start a slideshow. The `SlideshowWindow` is then displayed through the `SlideshowWindow#slideShowWindow.show()`.
+[NOTE]
+The slideshow window has already been instantiated on application launch and the window is merely being hidden or shown.
+
+*Step 3.* The window is now visible and existing questions are cleared. The list of slideshow questions is then fetched through
+`Logic#getSlideshowQuestions()` which in turn calls the `ModelManager#getSlideshowQuestions()` that fetches
+the `slideshowQuestions` list in `SavedQuestions`.
+
+*Step 4.* The user will then navigate and control the slideshow using the `Left/Right`, `Space` and `Esc` key as defined by the key
+listeners in `SlideshowWindow#initialiseKeyboardControls(Stage root)`.
+
+*Step 5.* The user exits the slideshow when the `Esc` key event is triggered or when the `currQuestionIndex` exceeds the `questionPanels.size()` or
+it falls below `0`. The behaviour of this follows the common procedure that most presentation programs adopt thus, it will not
+feel foreign to users.
+
+image::SlideshowFeatureActivityDiagram.png[]
```

Comment:
In the navigation loop, how is the current question incremented? As of now, your action box only mentions navigation, but does not specify directly how the current question counter is affected.
One possibility is to add another action box indicating the user incrementing/decrementing the question index